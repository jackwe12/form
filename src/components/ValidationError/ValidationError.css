.uk-text-danger{
    color:red;
    line-height: 21px;
}
.uk-text-danger{
    font-size: 12px;
}
.uk-form-danger{
    border:red solid 1px!important;
    box-shadow: 0 0 3px red !important;
    -moz-box-shadow: 0 0 3px red !important;
    -webkit-box-shadow: 0 0 3px red !important;
}

.uk-form-danger:focus{
    border:red solid 1px;
}