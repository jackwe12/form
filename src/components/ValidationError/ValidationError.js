import React from 'react';

const ValidationError = ({ content }) => {
    return <div className="uk-text-danger uk-text-small" style={{ "height": "21px" }}>{content}</div>
}

export default ValidationError;