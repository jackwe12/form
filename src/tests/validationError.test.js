import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import ValidationError from '../components/ValidationError/ValidationError';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

const wrapper = shallow(
    <ValidationError
        content="error"
/>)
const wrapper2 = shallow(
    <ValidationError
        content=""
/>)

describe('Testing <ValidationError/>', ()=>{
    test('snapshot testing',()=>{
        const tree =renderer
        .create(       <ValidationError
            content=""
    />).toJSON();
            expect(tree).toMatchSnapshot();
        })
    test('image tag should exist when alert',()=>{
        expect(wrapper.find('img').length).toBe(1);
        
    });
    test('image tag should not exist when no alert',()=>{
        expect(wrapper2.find('img').length).toBe(0);
    });
    test('warning should not exists when image is not clicked',()=>{
        const image = wrapper.find('img');
        expect(image.length).toBe(1);
        const div = wrapper.find('div');
        expect(div.text()).toEqual("");
    });
    test('warning should exists when image is clicked',()=>{
        const image = wrapper.find('img');
        expect(image.length).toBe(1);
        image.simulate('click');
        const div = wrapper.find('div');
        expect(div.text()).toEqual("error");
    });


});
