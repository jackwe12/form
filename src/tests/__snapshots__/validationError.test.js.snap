// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ValidationError/> snapshot testing 1`] = `
<div
  className="uk-text-danger"
  style={
    Object {
      "height": "21px",
    }
  }
>
  
  
</div>
`;
